test/main.sh

```diff
@@ -48,36 +48,77 @@ local_tcp_port() {
   done
 }
 
-# import all the backends
-for backend in backends/*.sh; do
-  # shellcheck disable=SC1090
-  . "${backend}"
-done
+# return a list of available storage backends
+available_storage_backends() {
+  # shellcheck disable=2039
+  local backend backends
+
+  backends="dir"
+  for backend in btrfs lvm zfs; do
+    if which $backend >/dev/null 2>&1; then
+      backends="$backends $backend"
+    fi
+  done
+  echo "$backends"
+}
+
+# whether a storage backend is available
+storage_backend_available() {
+  # shellcheck disable=2039
+  local backends
+  backends="$(available_storage_backends)"
+  [ "${backends#*$1}" != "$backends" ]
+}
+
+# choose a random available backend, excluding LXD_BACKEND
+random_storage_backend() {
+    # shellcheck disable=2046
+    shuf -e $(available_storage_backends | sed "s/\b$LXD_BACKEND\b//") | head -n 1
```

Comment:
The exclusion doesn't seem to be needed. The only thing you want to exclude is "random" but that's never going to be in available_storage_backends.